src/progpy/models/battery_electrochem.py

```diff
@@ -639,90 +652,272 @@ def merge_dicts(a: dict, b: dict) -> None:
+        kp : float
+            Lumped constant for BV (+ electrode)
+        Vol : float
+            Total interior battery volume/2 (for computing concentrations)
+        VolSFraction : float
+            Fraction of total volume occupied by surface volume
+        to : float
+            For Ohmic voltage
+        tsn : float
+            For surface overpotential (neg)
+        tsp : float
+            For surface overpotential (pos)
+        U0p : float
+            Redlich-Kister parameter (+ electrode)
+        Ap : float
+            Redlich-Kister parameter (+ electrode)
+        U0n : float
+            Redlich-Kister parameter (- electrode)
+        An : float
+            Redlich-Kister parameter (- electrode)
+        VEOD : float
+            End of discharge voltage threshold
+        VDropoff : float
+            Voltage above EOD after which voltage will be considered in SOC calculation
+        qMaxThreshold : float
+            Threshold for qMax (for threshold_met and event_state), after which the InsufficientCapacity event has occurred. Note: Battery manufacturers specify a threshold of 70-80% of qMax
+        wq : float
+            Wear rate for qMax
+        wr : float
+            Wear rate for Ro
+        wd : float
+            Wear rate for D
+        x0 : dict[str, float]
+            Initial :term:`state`
+
     See Also
     --------
     BatteryElectroChemEOL, BatteryElectroChemEOD, BatteryElectroChem
-
-    Note
-    ----
-    For keyword arguments, see BatteryElectroChemEOD, BatteryElectroChemEOL
     """
-    inputs = BatteryElectroChemEOD.inputs
-    outputs = BatteryElectroChemEOD.outputs
-    states = BatteryElectroChemEOD.states + BatteryElectroChemEOL.states
-    events = BatteryElectroChemEOD.events + BatteryElectroChemEOL.events
+    events = ['EOD', 'InsufficientCapacity']
+    inputs = ['i']
+    states = ['tb', 'Vo', 'Vsn', 'Vsp', 'qnB', 'qnS', 'qpB', 'qpS', 'qMobile', 'tDiffusion', 'Ro']
+    outputs = ['t', 'v']
+    performance_metric_keys = ['max_i']
 
     is_vectorized = False
 
-    default_parameters = deepcopy(BatteryElectroChemEOD.default_parameters)
-    merge_dicts(default_parameters,
-        BatteryElectroChemEOL.default_parameters)
+    state_limits = {
+        'tb': (0, np.inf),  # Limited by Absolute Zero (0 K)
+        'qnB': (0, np.inf),
+        'qnS': (0, np.inf),
+        'qpB': (0, np.inf),
+        'qpS': (0, np.inf),
+        'qMobile': (0, np.inf)
+    }
+
+    param_callbacks = {  # Callbacks for derived parameters
+        'VolSFraction': [update_vols, update_qpSBmin, update_qSBmax],
+        'Vol': [update_vols],
+        'xpMin': [update_qpSBmin],
+        'xpMax': [update_xnMin],
+        'xnMin': [update_qnmin],
+        'xnMax': [update_xpMin, update_qnmax, update_qnSBmax]
+    }
+
+    default_parameters = {  # Set to defaults
+        'xnMax': 0.6,
+        'xnMin': 0.0,
+        'xpMax': 1.0,
+        'xpMin': 0.4,
+        'qMax': 7600/0.6,
```

Comment:
I don't think we need qMax here anymore. Now it is calculated from qMobile as we go since qMobile is changing.

This means your helper function will have to receive qMax as a parameter now.

There are 2 derived x0 from qMax that I don't think you need anymore either. since it's dependent on another state those will then need to be calculated when you call init

We can talk more about this tomorrow

Reply:
Updated during sync - thanks Chris!